sqlglot/optimizer/merge_subqueries.py

```diff
@@ -119,6 +119,19 @@ def _mergeable(outer_scope, inner_select, leave_tables_isolated, from_or_join):
     Returns:
         bool: True if can be merged
     """
+
+    def _get_window_expressions_in_where_clause():
```

Comment:
I know there must be a cleaner way to do this...

Reply:
what exactly are you trying to do here?

Reply:
Get all the window expressions in the inner select scope. Then check if any of those columns are referenced by the outer scope but only in the outer scope's where clause (I guess I should also check join clause too). If so then block merge otherwise allow the merge.